message/validation/partial_validation.go

```diff
@@ -0,0 +1,116 @@
+package validation
+
+// partial_validation.go contains methods for validating partial signature messages
+
+import (
+	"fmt"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"github.com/herumi/bls-eth-go-binary/bls"
+
+	"github.com/bloxapp/ssv/protocol/v2/ssv/queue"
+	ssvtypes "github.com/bloxapp/ssv/protocol/v2/types"
+)
+
+func (mv *MessageValidator) validatePartialSignatureMessage(share *ssvtypes.SSVShare, msg *queue.DecodedSSVMessage) error {
+	signedMsg, ok := msg.Body.(*spectypes.SignedPartialSignatureMessage)
+	if !ok {
+		return fmt.Errorf("expected partial signature message")
+	}
+
+	if len(msg.Data) > maxPartialSignatureMsgSize {
+		return fmt.Errorf("size exceeded")
+	}
+
+	if err := mv.validPartialSigners(share, signedMsg); err != nil {
+		return err
+	}
+
+	// TODO: check running duty
+
+	if err := mv.validPartialSignatures(share, signedMsg); err != nil {
+		return err
+	}
+
+	role := msg.GetID().GetRoleType()
```

Comment:
We should also verify that the PartialSigMsgType is consistent with the role. This is duty specific rules, so:
- Attestation allows PostConsensusPartialSig
- Attestation Egg allows SelectionProofPartialSig and PostConsensusPartialSig
- Proposal allows RandaoPartialSig and PostConsensusPartialSig
- Sync Committee allows PostConsensusPartialSig
- Sync Committee Aggregation allows ContributionProofs and PostConsensusPartialSig
- ValidatorRegistration allows ValidatorRegistrationPartialSig